packages/desktop/views/dashboard/settings/views/profile/ProfileSettings.svelte

```diff
@@ -0,0 +1,33 @@
+<script lang="ts">
+    import { activeProfile } from '@core/profile'
+    import { ProfileSettingsRoute } from '@core/router'
+    import features from '@features/features'
+    import { HR } from 'shared/components'
+    import { ChangeProfileName, Currency, DeleteProfile } from './'
+
+    const settings: {
+        component: unknown
+        childRoute: ProfileSettingsRoute
+        requireLogin?: boolean
+    }[] = [
+        { component: ChangeProfileName, childRoute: ProfileSettingsRoute.ChangeProfileName },
+        { component: Currency, childRoute: ProfileSettingsRoute.Currency },
+        { component: DeleteProfile, childRoute: ProfileSettingsRoute.DeleteProfile },
+    ]
+    const visibleSettings = settings.filter((setting) => features?.settings?.profile?.[setting.childRoute]?.enabled)
```

Comment:
Although it's just a line, this could probably be made generic for any settings

Reply:
yeah this will be included when we refactor the settings into a generic component